File: 2021/R/day-07.R
x <- as.integer(strsplit(readLines("data/day-07.txt"), ",")[[1]])

day_07a <- function(x) {
  sum(abs(x - median(x)))
}

day_07b <- function(x) {
  ans <- Inf
  
  for (i in unique(x)) {
    res <- sum(unlist(sapply(abs(x - i), seq_len)))
    if (res < ans) ans <- res
  }
  
  ans
}


# test --------------------------------------------------------------------

stopifnot(
  day_07a(x) == 356958L,
  day_07b(x) == 105461913
)

File: 2021/Python/day-07.py
import numpy as np
import unittest


def day_07a(x) :
    x = None
    return x


def day_07b(x) :
    x = None
    return x


class testDay07(unittest.TestCase) :

    def test_day_07a(self) :
        self.assertEqual(day_07a(x), 356958)
  
    def test_day_07b(self) :
        self.assertEqual(day_07b(x), 105461913)
  

if __name__ == '__main__':
    x = np.loadtxt("data/day-7.txt", dtype=int, delimiter=",")
    unittest.main()
